package/rbd/rancher-rbd

```diff
@@ -0,0 +1,228 @@
+#!/bin/bash
+
+if [ -e "$(dirname $0)/common.sh" ]; then
+    source $(dirname $0)/common.sh
+elif [ -e "$(dirname $0)/../common/common.sh" ]; then
+    source $(dirname $0)/../common/common.sh
+fi
+DEVICE_TIMEOUT=10
+RBD_LOCK="has_mapped"
+
+init()
+{
+    print_success
+}
+
+create()
+{
+    # The OPTS variable will be populated from the input JSON as a map of key/values
+    if [ -z "${OPTS['name']}"  ]; then
+        print_error "name is required"
+    fi
+
+    local name=${OPTS['name']}
+    local size=${OPTS['size']:-"1G"}
+    local pool=${OPTS['pool']:-"rbd"}
+    local image_feature=${OPTS['image_feature']:-"layering"}
+    local OUT
+
+    OUT=$(rbd create ${pool}/${name} --size ${size} --image-feature ${image_feature})
+    if [ $? -ne 0  ]; then
+        print_error "${OUT}"
+    fi
+
+    print_options created true name ${name}
+}
+
+delete()
+{
+    # The OPTS variable will be populated from the input JSON as a map of key/values
+    if [ -z "${OPTS['name']}" ]; then
+        print_error "name is required"
+    fi
+
+    local name=${OPTS['name']}
+    local pool=${OPTS['pool']:-"rbd"}
+    local locker_id
+    local UNLOCK_OUT
+    local OUT
+
+    OUT=$(rbd info ${pool}/${name})
+    if [ $? -ne 0 ]; then
+        log_info ${pool}/${name} "Device does not exist:"
+        print_success
+        exit 0
+    fi
+
+    locker_id=$(rbd lock list ${pool}/${name} --format json | jq -c .${RBD_LOCK} | jq -r .locker)
+    if [ "${locker_id}" != "null" ]; then
+        UNLOCK_OUT=$(rbd lock remove ${pool}/${name} ${RBD_LOCK} ${locker_id})
+        log_info ${pool}/${name} "RBD remove lock err when deleting vol:"
+        print_success
```

Comment:
As Craig said, this one sounds dubious as well. Is it an error here? Also, `UNLOCK_OUT` wasn't used anywhere.